docs/guides/custom-http-client.mdx

```diff
@@ -0,0 +1,48 @@
+---
+id: custom-http-client
+title: Using a custom HTTP client (Experimental)
+description: Use a custom HTTP client for `sendRequest` and plain-HTTP crawling
+---
+
+import ApiLink from '@site/src/components/ApiLink';
+
+The <ApiLink to="core/basic/BasicCrawler">`BasicCrawler`</ApiLink> class allows you to configure the HTTP client implementation using the `httpClient` constructor option. This might be useful for testing or if you need to swap out the default implementation based on `got-scraping` for something else, such as `curl-impersonate` or `axios`.
+
+The HTTP client implementation needs to conform to the `BaseHttpClient` interface:
+
+```ts
+import {
+	BaseHttpClient,
+	HttpRequest,
+	HttpResponse,
+	RedirectHandler,
+	ResponseTypes,
+	StreamingHttpResponse,
+} from "@crawlee/core";
+
+class CustomHttpClient implements BaseHttpClient {
+	async sendRequest<TResponseType extends keyof ResponseTypes = "text">(
+		request: HttpRequest<TResponseType>,
+	): Promise<HttpResponse<TResponseType>> {
+        /* ... */
```

Comment:
can we use `fetch` or `axios` here or is that more tricky than it sounds like? the example without any implementation is not really helpful.

if the problematic part is streaming, maybe we could provide some default dummy implementation/helper that would just load things into memory and wrap it in a stream?

Reply:
I mean, if I'm going to implement it, I would rather just ship it as code. And writing an incomplete example is not exactly helpful either.

Reply:
What's wrong with doing both? Docs are important, you won't get around that just by publishing some package, we need them. If you don't want to do it now, let's show the got scraping client implementation here (which can be simplified, referencing the full working code), but we need something better than what we have now.

Reply:
Well, the curl-impersonate thing is (allegedly) slow and doesn't work on Windows - that's not something I'd like to put on display. A complete implementation of something else (fetch, axios, ...) would take some effort, and nobody asked for it (yet). That and the fact that we're going to change the interface pretty soon anyway (to be more ergonomic, among other things) kinda deter me.

Of course, I could just sketch something here and leave out the tricky parts (e.g.,  redirection + cookie handling), but that doesn't feel right either. It would look like we're leaving out the gnarly parts and leaving them for the user to figure out.

I guess just linking to that gist with curl-impersonate with a disclaimer could work?

Reply:
>  but that doesn't feel right either. 

It would still look better than what we have now. I am fine with leaving out tricky parts for now, but leaving out everything feels weird.

Btw, people already asked indirectly about axios support, we have several reports about things not working with got scraping, but working with axios. 

A link to that gist could also help, and I would surely add it here.